http/sys_leader.go

```diff
@@ -17,19 +17,29 @@ func handleSysLeader(core *vault.Core, opt ...ListenerConfigOption) http.Handler
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		switch r.Method {
 		case "GET":
-			handleSysLeaderGet(core, w, opt...)
+			handleSysLeaderGet(core, w, r, opt...)
 		default:
 			respondError(w, http.StatusMethodNotAllowed, nil)
 		}
 	})
 }
 
-func handleSysLeaderGet(core *vault.Core, w http.ResponseWriter, opt ...ListenerConfigOption) {
-	ctx := context.Background()
+func handleSysLeaderGet(core *vault.Core, w http.ResponseWriter, r *http.Request, opt ...ListenerConfigOption) {
+	var tokenPresent bool
+	token := r.Header.Get(consts.AuthHeaderName)
+	ctx := r.Context()
+
+	if token != "" {
+		// We don't care about the error, we just want to know if token exists
+		lock := core.HALock()
+		lock.Lock()
+		tokenEntry, err := core.LookupToken(ctx, token)
+		lock.Unlock()
+		tokenPresent = err == nil && tokenEntry != nil
+	}
 
-	opts, _ := getOpts(opt...)
-	if opts.withRedactAddresses {
-		ctx = logical.CreateContextRedactionSettings(ctx, false, true, false)
+	if tokenPresent {
+		ctx = logical.CreateContextRedactionSettings(r.Context(), false, false, false)
```

Comment:
So the reason this works is because of the `redactionSettingsWrapping` method; this will wrap this with the correct configuration values before we even get to sys_leader. Same with `sys_seal`

In my ENT PR I have end to end tests that verify this as well.

Reply:
The correct context values are actually already set in `redactionSettingsWrapping`. I have some end to end tests for this in the ENT PR.

Reply:
I'm sorry, you're absolutely correct. I can see it now.